### PR TITLE
Feature/script packing

### PR DESCRIPTION
**Pull request by ProofOfKeags**
This PR adds the ability to pack scripts into the s9pk similarly to how assets are packed. It also adds the ability to unpack them. There is currently NO VALIDATION that the scripts are A. scripts at all B. scripts that are actual js or C. that they conform to the right type needed for execution.

### CODE REVIEW

**Comment by ProofOfKeags** [comment]
This is ready for review now

**Comment by ProofOfKeags** [comment]
bernie-PR

File: backend/src/install/cleanup.rs [hunk 1]
```diff
@@ -148,6 +148,28 @@ pub async fn cleanup(ctx: &RpcContext, id: &PackageId, version: &Version) -> Res
             .await
             .apply(|res| errors.handle(res));
     }
+    let assets_path = ctx
+        .datadir
+        .join(PKG_VOLUME_DIR)
+        .join(id)
+        .join("assets")
+        .join(version.as_str());
```
Comment (dr-bonez) [inline backend/src/install/cleanup.rs:156]: I think there's a function `assets_dir` in volumes.rs that would be better to use for this

File: backend/src/install/cleanup.rs [hunk 2]
```diff
@@ -148,6 +148,28 @@ pub async fn cleanup(ctx: &RpcContext, id: &PackageId, version: &Version) -> Res
...
+        tokio::fs::remove_dir_all(&assets_path)
+            .await
+            .apply(|res| errors.handle(res));
+    }
+    let scripts_path = ctx
+        .datadir
+        .join(PKG_VOLUME_DIR)
+        .join(id)
+        .join("scripts")
+        .join(version.as_str());
```
Comment (dr-bonez) [inline backend/src/install/cleanup.rs:167]: This path is incorrect

File: backend/src/s9pk/header.rs [hunk 3]
```diff
@@ -93,6 +95,10 @@ impl TableOfContents {
         self.docker_images
             .serialize_entry("docker_images", &mut writer)?;
         self.assets.serialize_entry("assets", &mut writer)?;
+        match self.scripts {
+            None => FileSection::default().serialize_entry("scripts", &mut writer),
+            Some(a) => a.serialize_entry("scripts", &mut writer),
+        };
```
Comment (dr-bonez) [inline backend/src/s9pk/header.rs:101]:
```suggestion
    self.scripts.unwrap_or_default().serialize_entry("scripts", &mut writer)?;
```

**Review by dr-bonez — CHANGES_REQUESTED** [review]

**Review by dr-bonez — APPROVED** [review]

**Pull request merged**